### DISCUSSION TITLE
Icon not appearing in MenuFlyoutItem

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14214] [Q&A]
This could take a while, so please bear with me.

In my main XAML file I have
```xaml
            <MenuBarItem Title="Settings" BorderBrush="LightGray"  BorderThickness="1" x:Name="mnuSettings" >
                <MenuFlyoutItem x:Name="mnuIDEConfiguration" Text="Editor Configuration" Click="mnuIDEConfiguration_Click"  Command="{Binding GoToIDEConfig}" />
                <MenuFlyoutSubItem x:Name="mnuSelectLanguage" Text="Choose interface language"></MenuFlyoutSubItem>
            </MenuBarItem>
```
The reason that mnuSelectLanguage is a MenuFlyoutSubItem is because I want to populate it with MenuFlyoutItems, viz
```cs
                    MenuFlyoutItem menuFlyoutItem = new()
                    {
                        Text = globals[$"{100 + i + 1}"],
                        Name = names.First(),  
                    };
                    menuFlyoutItem.Click += routedEventHandler; //  Internationalization_Click;
                    menuBarItem.Items.Add(menuFlyoutItem);
```
And this works. I end up with a menu that looks like this:
![image](https://github.com/unoplatform/uno/assets/209621/07f4d8fa-3c09-424e-8989-e3441735541e)
So having built the menu, I then select an item. I try to set the .Icon of the element to a tick.

However, whilst the setting of the icon doesn't throw any error, it doesn't actually show the icon. 
```cs
        private void Internationalization_Click(object sender, RoutedEventArgs e)
        {
            FontIcon tickIcon = new FontIcon()
            {
                FontFamily = new FontFamily("Segoe MDL2 Assets"),
                Glyph = "\uF0B7"
            };

            var me = (MenuFlyoutItem)sender;
            var name = me.Name;

            foreach (var item in mnuSelectLanguage.Items)
            {
                ((MenuFlyoutItem)item).Icon = item.Name == name ? tickIcon : (IconElement?)null;
            }

            HandleLanguageChange(name);
        }
```
(On a side note, I'm hoping someone shows me how to convert tickIcon to a XAML resource but that's a distraction right now.)

Now, I've seen in other places the idea that Icon's don't appear in certain circumstances like say a StackPanel. Nothing like that's happening here. 

So what's obviously wrong with my code? Why can't I set the Icon and have it appear?

BTW, not fixed by upgrading to Uno 5.0

-Bruce

**Comment by jeromelaban** [comment]
This looks like a bug, can you open an issue with a repro? Thanks!
